### PR TITLE
[WIP] Building content-api with rollup

### PR DESCRIPTION
**Pull request by allouis**
no-issue

Mostly following the development here: https://github.com/TryGhost/Ghost-SDK/pull/31

TODO:
- [x] update tests to work with ES modules or CJS build (used CJS build)
- [x] add prepublish command to build files
- [x] ignore built files
- [x] test in gatsby-starter (es6)
- [x] test in node
- [x] test UMD build

### CODE REVIEW

File: packages/content-api/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                preferBuiltins: false
+            }),
+            commonjs({
+                include: 'node_modules/**'
+            }),
+            babel({
+                exclude: 'mode_modules/**'
+            }),
+            replace({
+                'process.env.NODE_ENV': `"${process.env.NODE_ENV}"`
```
Comment (kevinansfield) [inline packages/content-api/rollup.config.js]: Why was this needed? I couldn't see it used in the package source, something in axios?

When the package is published what would the value of NODE_ENV be? I wonder if it needs to be set to a fixed value for the UMD build 🤔

File: packages/content-api/package.json [hunk 2]
```diff
@@ -4,25 +4,34 @@
...
+  "source": "lib/index.js",
   "scripts": {
     "dev": "echo \"Implement me!\"",
+    "pretest": "yarn build",
     "test": "NODE_ENV=testing mocha './test/**/*.test.js'",
-    "prebuild": "rm -rf build && mkdir build",
-    "build": "browserify -t varify -g es6ify -s ContentApi index.js > build/content-api.lib.js",
-    "lint": "eslint . --ext .js --cache",
+    "build": "rollup -c",
+    "lint": "eslint ./src --ext .js --cache",
```
Comment (kevinansfield) [inline packages/content-api/package.json]: Should be swapped too `./lib` or follow https://github.com/TryGhost/Ghost-SDK/pull/31/commits/ca0f07afe86bd28a638e585731a6091a6863e28f to only ignore `dist`

File: packages/content-api/rollup.config.js (hunk 1 above)
Comment (allouis) [reply packages/content-api/rollup.config.js]: Yeah - something in `axios` was testing against `process.env.NODE_ENV` - In the prepublish command i've explicitly set it to 'production' :relaxed:
